packages/@ember/-internals/runtime/types/observable.d.ts

```diff
@@ -0,0 +1,81 @@
+import { ObserverMethod } from './-private/types';
+
+/**
+ * This mixin provides properties and property observing functionality, core features of the Ember object model.
+ */
```

Comment:
A discussion point (cc. @pzuraq @dfreeman @jamescdavis): throughout, this uses the `keyof this` to make keys support both auto-complete and type-checking against property keys. However, it *drops* the feature from DT where it returns the type. I *think*, if we drop support for building types mixins, we can actually enable `strictFunctionTypes` and keep the return type correct?